components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
                 if (claimsMap.containsKey(SCIMConstants.USER_NAME_URI)) {
                     claimsMap.remove(SCIMConstants.USER_NAME_URI);
                 }
-                carbonUM.addUser(user.getUserName(), user.getPassword(), null, claimsMap, null);
+
+                if (claimsMap.containsKey(SCIMConstants.META_LOCATION_URI)) {
+                    claimsMap.remove(SCIMConstants.META_LOCATION_URI);
+                }
+
+                Map<String, String> clonedClaimsMap = new HashMap(claimsMap);
+                carbonUM.addUser(user.getUserName(), user.getPassword(), null, clonedClaimsMap, null);
+                //Remove username from the modified map before compare with the original map
```

Comment:
Add the **reason** to remove modified username.

Reply:
Add a new line before the comment.

And add a space before starting the text. ie.
`// Remove`